pkg/controller/backupbucket/backupbucket.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2023 OnMetal authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backupbucket
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	controllerconfig "github.com/onmetal/gardener-extension-provider-onmetal/pkg/apis/config"
+	"github.com/onmetal/gardener-extension-provider-onmetal/pkg/onmetal"
+	storagev1alpha1 "github.com/onmetal/onmetal-api/api/storage/v1alpha1"
+)
+
+// ensureBackupBucket creates onmetal backupBucket object and returns access to bucket
+func (a *actuator) ensureBackupBucket(ctx context.Context, namespace string, onmetalClient client.Client, backupBucket *extensionsv1alpha1.BackupBucket) error {
+	bucket := &storagev1alpha1.Bucket{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      backupBucket.Name,
+			Namespace: namespace,
+		},
+		Spec: storagev1alpha1.BucketSpec{
+			BucketClassRef: &corev1.LocalObjectReference{
+				Name: a.backupBucketConfig.BucketClassName,
+			},
+		},
+	}
+	//create onmetal bucket
+	if _, err := controllerutil.CreateOrPatch(ctx, onmetalClient, bucket, nil); err != nil {
+		return fmt.Errorf("failed to create or patch backup bucket %s: %w", client.ObjectKeyFromObject(bucket), err)
+	}
+	//wait for bucket creation
+	if err := wait.PollUntilWithContext(ctx, 1*time.Second, func(ctx context.Context) (done bool, err error) {
```

Comment:
Can we use here a backoff polling like we are doing in the CCM: https://github.com/onmetal/cloud-provider-onmetal/blob/main/pkg/cloudprovider/onmetal/load_balancer.go#L187

Reply:
Modified to use wait.ExponentialBackoffWithContext